### PR TITLE
Support for Elasticsearch 6.*

### PR DESCRIPTION
**Pull request by JasparGupta**
Ok, the only thing I'm not 100% on is in the `Connection::class` in the `statement()` function, it doesn't quite feel right to be there, but I guess thats because it was meant for SQL where you can have a SQL statement that does many things.

But yeah, so you can create a migration file that will sit in the `migrations` directory and will be run with the native `php artisan migrate` command, and it will either create the new mapping and alias it if the alias doesn't already exists, or it will update an existing mapping using the alias to identify the index.

### CODE REVIEW

File: src/Database/Schema/Blueprint.php [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+use Illuminate\Database\Connection;
+use Illuminate\Database\Schema\Grammars\Grammar;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Str;
+
+/**
+ * Class Blueprint
+ * @package DesignMyNight\Elasticsearch\Database\Schema
+ */
+class Blueprint extends \Illuminate\Database\Schema\Blueprint
```
Comment (willtj) [inline src/Database/Schema/Blueprint.php]: Are all the column types here ES-specific? I don't fully understand the 'range' ones

**Comment by JasparGupta** [comment]
Seems I forgot to support dynamic mapping thought...

File: src/Database/Schema/Blueprint.php (hunk 1 above)
Comment (JasparGupta) [reply src/Database/Schema/Blueprint.php]: Yeah I copied them from here: https://www.elastic.co/guide/en/elasticsearch/reference/6.7/mapping-types.html

Comment (JasparGupta) [reply src/Database/Schema/Blueprint.php]: And the range sub types they have here: https://www.elastic.co/guide/en/elasticsearch/reference/6.7/range.html

**Comment by JasparGupta** [comment]
Yeahhhh I guess I could add some tests

File: tests/Database/Schema/Grammars/ElasticsearchGrammarTest.php [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+                        'type' => 'text'
+                    ],
+                    'coordinates' => [
+                        'type' => 'geo_point'
+                    ]
+                ]
+            ]
+        ];
+
+        $grammar = new class extends ElasticsearchGrammar
```
Comment (JasparGupta) [inline tests/Database/Schema/Grammars/ElasticsearchGrammarTest.php]: Do you like my use of anonymous classes to test protected methods!? 😀

File: src/Database/Schema/PropertyDefinition.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace DesignMyNight\Elasticsearch\Database\Schema;
+
+use Illuminate\Support\Fluent;
```
Comment (willtj) [inline src/Database/Schema/PropertyDefinition.php]: I never knew `Fluent` existed - this is great!

File: tests/Database/Schema/Grammars/ElasticsearchGrammarTest.php (hunk 2 above)
Comment (willtj) [inline tests/Database/Schema/Grammars/ElasticsearchGrammarTest.php]: So clever!

File: src/Database/Schema/PropertyDefinition.php (hunk 3 above)
Comment (JasparGupta) [reply src/Database/Schema/PropertyDefinition.php]: Yeah one of those magic little gems hidden within the framework, such a cool little idea to populate a key/value array!

**Pull request merged**